test/scripts/parallel-local/test-runner.js

```diff
@@ -0,0 +1,101 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+let isFirstTest = true;
+
+/**
+ * Ensures the directory exists before executing tests.
+ * @param {string} testType The type of the test.
+ * @returns {void} Does not return a value.
+ */
+const ensureDirectoryExists = (testType) => {
+    const dirPath = path.join('./database/storage', testType);
+    if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath, { recursive: true });
+    }
+};
+
+/**
+ * Handles incoming messages to determine test execution workflow.
+ * @param {string|Object} message - Message received from the main process.
+ * @returns {void} Does not return a value.
+ */
+const processMessage = (message) => {
+    if (message === 'no_more_tests') {
+        process.exit();
+    } else {
+        manageTestExecution(message);
+    }
+};
+
+process.on('message', processMessage);
+
+/**
+ * Executes a set of tests based on received settings.
+ * @param {Object} testConfig - Configuration for the test.
+ * @returns {void} Does not return a value.
+ */
+const manageTestExecution = (testConfig) => {
```

Comment:
Avoid abbreviation, it should be `testConfiguration`

Reply:
Done